PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,476 @@
+// -*- C++ -*-
+//
+// Package:    ProdTutorial/L1PrefiringWeightProducer
+// Class:      L1PrefiringWeightProducer
+//
+/**\class L1PrefiringWeightProducer L1PrefiringWeightProducer.cc ProdTutorial/L1ECALPrefiringWeightProducer/plugins/L1ECALPrefiringWeightProducer.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  localusers user
+//         Created:  Thu, 08 Nov 2018 16:16:00 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+
+#include "TFile.h"
+#include "TF1.h"
+#include "TH2.h"
+
+#include <iostream>
+enum fluctuations { central = 0, up, down, upStat, downStat, upSyst, downSyst };
+
+class L1PrefiringWeightProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1PrefiringWeightProducer(const edm::ParameterSet&);
+  ~L1PrefiringWeightProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  double getPrefiringRateEcal(double eta, double pt, TH2F* h_prefmap, fluctuations fluctuation) const;
+  double getPrefiringRateMuon(double eta, double phi, double pt, fluctuations fluctuation) const;
+
+  const edm::EDGetTokenT<std::vector<pat::Photon> > photons_token_;
+  const edm::EDGetTokenT<std::vector<pat::Jet> > jets_token_;
+  const edm::EDGetTokenT<std::vector<pat::Muon> > muon_token_;
+
+  const edm::EDPutTokenT<double> nonPrefiringProbToken_;
```

Comment:
@slava77 I was asked to switch the type produced by this producer to float for nanoAOD. I changed it in the nanoAOD config (see https://github.com/cms-sw/cmssw/pull/33759/files#diff-9b898944f08810913ea5c1605c47bd9a6c99eb91fc20f491ba2767702f4a0184L226-R252), but not in the producer, which currently leads to the test failing. If I change it here, it will change the type also for the input to miniAOD, which from your previous concerns I understand is not allowed?

Reply:
The current test will still fail, btw, you might want to abort it.

Reply:
> If I change it here, it will change the type also for the input to miniAOD, which from your previous concerns I understand is not allowed?

I am still confused by "will change the type also for the input to miniAOD", because the input to miniAOD is AOD.

Assuming we are talking about the output of miniAOD, then yes, changing the collection type is not allowed in the production miniAOD.

Reply:
Sorry, yes, I meant the output. I have reverted everything to doubles and now it should adhere to the rules.